### ISSUE TITLE
[ Linux ] Mods installed from vault create a secondary flattened directory structure

### ISSUE THREAD

**Issue by douglasg14b** [FAForever/downlords-faf-client#1208]
When a mod is installed from the vault, that mod is put int the Mods directory as you would expect, but the entire directory structure of that mod will also be flattened into the Mods folder with the full path of the structure being the name of the folder. Creating a huge mess of empty folders.

This also causes the client to show that mods are corrupt for every one of these folders in the notifications.

Example:

![image](https://user-images.githubusercontent.com/1400380/57262021-41198780-701f-11e9-874d-8607b4df71d2.png)

**Comment by tatsujb** [comment]
this is if maps and mods sim-linking isn't in effect : 

```
cd ~/.steam/steam/steamapps/common/Supreme\ Commander\ Forged\ Alliance
rm Maps
rm Mods
ln -s ~/My\ Games/Gas\ Powered\ Games/Supreme\ Commander\ Forged\ Alliance/Maps/ Maps
ln -s ~/My\ Games/Gas\ Powered\ Games/Supreme\ Commander\ Forged\ Alliance/Mods/ Mods
cd ~/.steam/steam/steamapps/compatdata/9420/pfx/drive_c/users/steamuser
rm -rf My\ Documents
mkdir My\ Documents
cd My\ Documents
ln -s ~/My\ Games/ My\ Games
```

**Comment by tatsujb** [comment]
still happens for some maps.

**Comment by 1-alex98** [comment]
That's because \ is for windows and / is for unix i guess

**Comment by 1-alex98** [comment]
Can u pls check if that is the case in the version > v1.0.0 and in old versions, we indeed changed code between that versions

**Comment by 1-alex98** [comment]
oh this issue is damn old i thought it was new

**Comment by tatsujb** [comment]
hmm maybe it's fixed.
only mod I could still reproduce it for was quality of performance but now it's not in the vault anymore.
(better off)

**Comment by tatsujb** [comment]
nope. people still have this mod and host games with it and when you join it still does this : 
+"TheDairyFarm" whatever that is : 
![Screenshot from 2019-12-18 18-58-44](https://user-images.githubusercontent.com/5132359/71110891-76e33900-21c8-11ea-9372-dbbe95f40201.png)

**Comment by Slothologist** [comment]
It seems this is actually not a bug of the client, but an issue with the zip files/ mods themselves. Upon inspection, https://content.faforever.com/mods/the_dairy_farm.v0005.zip which is downloaded when installing this mod, always contains these weird folders, **_regardless of the archive manager used on linux_.** This is not the case for other mods, such as https://content.faforever.com/mods/supreme_score_board.v0002.zip
Could a Windows user confirm whether or not those weird folders are created/included in the archive on your OS as well? It might just be that Windows does not unzip these folders or ignores them for whatever reason. Or that Linux has some problem with these specific archives.
You will also note that TheDairyfarm gets correctly installed, and can be played. I would suspect the same for Quality of performance (but cannot test that atm).

**Comment by 1-alex98** [comment]
Really old mods were packed by the py client, but since a long time that is the job of the java client. So it might be that the time difference between the both make some mods linux compatible and others not.

**Comment by 1-alex98** [comment]
![grafik](https://user-images.githubusercontent.com/19996713/75909504-83cd0b80-5e4c-11ea-8423-fdfe4ce62a74.png)
No flat directory structure on windows, so probabaly / vs \

**Comment by 1-alex98** [comment]
Also we could try to hack in that it works for both OSES but this then is a java-commons issue
